services/ui_backend_service/api/autocomplete.py

```diff
@@ -0,0 +1,110 @@
+from services.utils import handle_exceptions, web_response
+import asyncio
+import threading
+
+
+class AutoCompleteApi(object):
+    # Cache tags so we don't have to request DB everytime
+    tags = []
+
+    def __init__(self, app, db):
+        self.db = db
+        # Cached resources
+        app.router.add_route("GET", "/autocomplete/tags", self.get_all_tags)
+        # Non-cached resources
+        app.router.add_route("GET", "/autocomplete/flows", self.get_all_flows)
+        app.router.add_route("GET", "/autocomplete/flows/{flow_id}/runs", self.get_runs_for_flow)
+        app.router.add_route("GET", "/autocomplete/flows/{flow_id}/runs/{run_id}/steps", self.get_steps_for_run)
```

Comment:
opinions on routing, feel free to disagree: should we stick to the original restful paths and instead have the `autocomplete` as a trailing resource? i.e. `/flows/{flow_id}/runs/autocomplete` as it is a query that is targeted at a resourcepath, instead of "autocomplete" being a distinct resource?

Reply:
/flows/autocomplete does clash with `"/flows/{flow_id}"` etc but maybe i can prioritize autocomplete endpoints first

Reply:
mm that's true.. and can't have a flow with the name autocomplete then. good point